rules/integrations/endpoint/defense_evasion_elastic_memory_threat_detected.toml

```
@@ -0,0 +1,178 @@
+[metadata]
+creation_date = "2024/03/24"
+integration = ["endpoint"]
+maturity = "production"
+min_stack_comments = "New fields added: required_fields, related_integrations, setup"
+min_stack_version = "8.16.0"
+promotion = true
+updated_date = "2024/11/26"
+
+[rule]
+author = ["Elastic"]
+description = """
+Generates a detection alert each time an Elastic Defend alert for memory signatures are received. Enabling this rule
+allows you to immediately begin investigating your Endpoint memory signature alerts. This rule identifies Elastic Defend
+memory signature detections only, and does not include prevention alerts.
+"""
+enabled = true
+from = "now-10m"
+index = ["logs-endpoint.alerts-*"]
+interval = "5m"
+language = "kuery"
+license = "Elastic License v2"
+max_signals = 10000
+name = "Memory Threat - Detected - Elastic Defend"
+note = """## Triage and analysis
+
+### Investigating Memory Threat Alerts
+
+Elastic Endpoint’s memory threat protection adds a layer of coverage for advanced attacks which avoid the traditional approach of writing payloads to disk. Instead, the malicious code runs only in-memory, an effective technique for evading legacy security products. There are currently two sub-categories of memory threat protection.
+
+The first category is referred to as memory signatures and is available on all supported OS. It operates by periodically scanning process executable memory regions based on their activity to identify and terminate known bad malware.
+
+The second category is referred to as shellcode thread and is unique to Windows endpoints today. A common technique of in-memory malware is to load the payload in a memory region not backed by a file on disk and create a thread to execute it.
+
+
+### Possible investigation steps
+
+- Investigate the process execution chain (parent process tree) for unknown processes. Examine their executable files for prevalence, whether they are located in expected locations, and if they are signed with valid digital signatures.
+- Assess whether this behavior is prevalent in the environment by looking for similar occurrences across hosts :
+   - For shellcode alerts, the key for bucketing alerts is stored in the `Memory_protection.unique_key_v1` field.
+   - For Memory signature alerts are bucketed based on the signatures which hit `rule.name`.
```

```suggestion
   - For Memory signature alerts, bucket based on the signatures which match `rule.name`.
```

Not sure if this change still captures the original meaning, but if so we can copy to other rules with the same sentence.